pydarn/plotting/rtp.py

```diff
@@ -0,0 +1,988 @@
+                    # else plot missing data
+                    elif len(x) > 0:
+                        diff_time = time - x[-1]
+                        if diff_time.seconds/60 > 2.0:
+                            x.append(time)
+                            y.append(np.nan)  # for masking the data
+            # Check if there is any data to plot
+            if np.all(np.isnan(y)) or len(x) == 0:
+                raise rtp_exceptions.RTPNoDataFoundError(parameter, beam_num,
+                                                         start_time, end_time,
+                                                         cls.dmap_data[0]['bmnum'])
+
+            # using masked arrays to create gaps in the plot
+            # otherwise the lines will connect in gapped data
+            my = np.ma.array(y)
+            my = np.ma.masked_where(np.isnan(my), my)
+            lines = ax.plot_date(x, my, fmt='k', tz=None, xdate=True,
+                                 ydate=False,
+                                 **kwargs)
+            rounded_down_start_time = x[0] - timedelta(minutes=x[0].minute % 15,
+                                      seconds=x[0].second,
+                                      microseconds=x[0].microsecond)
+            ax.set_xlim([rounded_down_start_time, x[-1]])
+            ax.set_yscale(scale)
+        # set date format and minor hourly locators
+        # Rounded the time down to show origin label upon
+        # Daniel Billet and others request.
+        # TODO: may move this to its own function
+        rounded_down_start_time = x[0] - timedelta(minutes=x[0].minute % 15,
+                                  seconds=x[0].second,
+                                  microseconds=x[0].microsecond)
+        ax.set_xlim([rounded_down_start_time, x[-1]])
+
+        ax.xaxis.set_major_formatter(dates.DateFormatter(date_fmt))
+        ax.xaxis.set_minor_locator(dates.HourLocator())
+        ax.margins(x=0)
+        ax.tick_params(axis='y', which='minor')
+        return lines, x, y
+
+    @classmethod
+    def plot_summary(cls, dmap_data: List[dict], beam_num: int = 0,
+                     groundscatter: bool = False, channel: int = 'all',
+                     figsize: tuple = (11, 8.5), boundary: dict = {},
+                     background_color: str = 'w',
+                     cmaps: dict = {}, lines: dict = {},
+                     plot_elv: bool = True, title=None):
+        """
+        Plots the summary of the following SuperDARN parameter plots:
+            - noise.search : (time-series)
+            - noise.sky :  (time-series)
+            - tfreq : transmission frequency (time-series)
+            - nave : number of averages  (time-series)
+            - cp : control program ID (time-series)
+            - p_l : Signal to Noise ratio (range-time)
+            - v : velocity (range-time)
+            - w_l : spectral width (range-time)
+            - elv : elevation (optional) (range-time)
```

Comment:
This should all be in a notes section on the bottom (or in the returns description).

Reply:
I think that is a personal preference. Personally I like it at the top as it details exactly what is being plotted as a descriptor should say and when users look at the docstring this would be the first thing they see. Where Notes would require some scrolling.

Reply:
From the numpy style guide, the first line should be a: " one-line summary that does not use variable names or the function name"

Reply:
Those are not variable names for summary plots and we don't need to be super strict with this. I think this classifies as high conflict vs. low reward. However, I don't want to hold up the merge any longer, so I will add this to the Notes as per request.

Reply:
Done.